tests/prototypes/prototype_kafka.py

```diff
@@ -0,0 +1,395 @@
+    kafka_topic: KafkaTopic
+    consumer_cxt: ConsumerContextManager
+    num_frames: NumFrames
+    frame_rate: FrameRate
+
+    def merge_bytes(self, datalist: list[bytes]) -> bytes:
+        from functools import reduce
+
+        return reduce(lambda x, y: x + y, datalist)
+
+    @retry(Empty, max_trials=10, interval=0.1)
+    def _poll(self, consumer: Consumer) -> Message:
+        if (msg := consumer.poll(timeout=0)) is None:
+            raise Empty
+        else:
+            return msg
+
+    def poll(self, consumer: Consumer) -> Message | None:
+        try:
+            return self._poll(consumer)
+        except Empty:
+            return None
+
+    @abstractmethod
+    def poll_one_data(self, consumer: Consumer) -> sc.DataArray | None:
+        ...
+
+    async def send_data_chunk(self, data_chunk: Events) -> None:
+        self.debug(
+            "Sending %s th, %s pieces of data.", self.data_counts, len(data_chunk)
+        )
+        self.raw_data_pipe.append(Events(data_chunk))
+        await self.commit_process()
+
+    def start_stop_watch(self) -> None:
+        self.stop_watch.start()
+
+    async def run(self) -> None:
+        with self.consumer_cxt() as consumer:
+            self.start_stop_watch()
+            data_chunk: Events = Events([])
+            while (event := self.poll_one_data(consumer)) is not None:
+                data_chunk.append(event)
+                if len(data_chunk) >= self.chunk_size:
+                    await self.send_data_chunk(data_chunk)
+                    data_chunk = Events([])
+
+            if data_chunk:
+                await self.send_data_chunk(data_chunk)
+
+            self.info("Data streaming finished...")
+
+
+class KafkaListenerScippOnly(KafkaListenerBase):
+    def poll_one_data(self, consumer: Consumer) -> sc.DataArray | None:
+        import json
+
+        from scipp.serialization import deserialize
+
+        header: dict = {}
+        data_list: list[bytes] = []
+        header_prefix = b'header'
+        finished_prefix = b'finished'
+
+        while msg := self.poll(consumer):
+            raw_msg: bytes = msg.value()
+            if raw_msg.startswith(header_prefix):
+                header = json.loads(raw_msg.removeprefix(header_prefix))
+            elif raw_msg.startswith(finished_prefix):
+                da = deserialize(header, [self.merge_bytes(data_list)])
+                if not isinstance(da, sc.DataArray):
+                    raise TypeError('Expected sc.DataArray, but got ', type(da))
+                return da
+            else:
+                data_list.append(raw_msg)
+
+        return None
+
+
+class KafkaListener(KafkaListenerBase):
+    def deserialize(self, data_list: list[bytes]) -> sc.DataArray:
+        import numpy as np
+        from streaming_data_types.eventdata_ev44 import deserialise_ev44
+
+        data = deserialise_ev44(self.merge_bytes(data_list))
+        event_zeros = np.full(len(data.pixel_id), data.reference_time[0])
```

Comment:
Initializing a full numpy array and copying into a variable is a bit inefficient, should directly init a scipp variable. Use `scipp.full`?